File: frontend/src/types/incomingDelivery.ts
export interface IncomingDeliveryItem {
  id: number;
  deliveryId: number;
  skuId: number | null;
  article: string;
  name: string | null;
  barcode: string | null;
  quantity: number;
  factQuantity: number | null;
  weight: string | null;
  volume: string | null;
  sku: { id: number; article: string; name: string } | null;
}

export interface IncomingDelivery {
  id: number;
  number: string;
  partnerId: number;
  warehouseCode: string | null;
  isCrossDock: boolean;
  status: string;
  plannedDate: string | null;
  actualDate: string | null;
  comment: string | null;
  createdBy: string;
  createdAt: string;
  updatedAt: string;
  partner: { id: number; name: string };
  items: IncomingDeliveryItem[];
}

export interface IncomingDeliveriesResponse {
  data: IncomingDelivery[];
  total: number;
}

export interface IncomingDeliveryItemInput {
  id?: number;
  article: string;
  name?: string;
  barcode?: string;
  quantity: number;
  weight?: number;
  volume?: number;
}

export interface IncomingDeliveryFormData {
  partnerId: number;
  warehouseCode?: string;
  isCrossDock?: boolean;
  plannedDate?: string;
  comment?: string;
  items: IncomingDeliveryItemInput[];
}

export interface StatusChangeEntry {
  id: number;
  docType: string;
  docId: number;
  status: string;
  changedBy: string;
  changedAt: string;
}

/** Создана → Процесс (первая операция в ТСД/интерфейсе) → Выполнено; Отмена — вручную. Общая схема для ВХП/ИСП/Инвентаризации. */
export const DOCUMENT_STATUSES = ['Создана', 'Процесс', 'Выполнено', 'Отмена'] as const;

export function deliveryWeight(delivery: IncomingDelivery): number {
  return delivery.items.reduce((sum, i) => sum + (i.weight != null ? Number(i.weight) : 0), 0);
}

export function deliveryVolume(delivery: IncomingDelivery): number {
  return delivery.items.reduce((sum, i) => sum + (i.volume != null ? Number(i.volume) : 0), 0);
}

File: mobil/core/model/src/main/java/com/npp/tsd/core/model/IncomingDelivery.kt
package com.npp.tsd.core.model

import kotlinx.serialization.Serializable

@Serializable
data class IncomingDeliveryItem(
    val id: Int,
    val deliveryId: Int,
    val skuId: Int? = null,
    val article: String,
    val name: String? = null,
    val barcode: String? = null,
    val quantity: Int,
    val factQuantity: Int? = null,
    val weight: String? = null,
    val volume: String? = null,
    val sku: SkuRef? = null,
)

@Serializable
data class IncomingDelivery(
    val id: Int,
    val number: String,
    val partnerId: Int,
    val warehouseCode: String? = null,
    val isCrossDock: Boolean = false,
    val status: String,
    val plannedDate: String? = null,
    val actualDate: String? = null,
    val comment: String? = null,
    val createdBy: String,
    val createdAt: String,
    val updatedAt: String,
    val partner: PartnerRef,
    val items: List<IncomingDeliveryItem> = emptyList(),
)

@Serializable
data class IncomingDeliveriesResponse(
    val data: List<IncomingDelivery>,
    val total: Int,
)

@Serializable
data class ReceiveIncomingDeliveryItemBody(
    val itemId: Int,
    val factQuantity: Int,
)

@Serializable
data class ReceiveIncomingDeliveryBody(
    val items: List<ReceiveIncomingDeliveryItemBody>,
)
